src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -192,79 +144,56 @@ final case class PurposeApiServiceImpl(
       enhancedPurpose <- enhancePurpose(updatedPurpose, eService, ownership)
     } yield enhancedPurpose
 
-    val defaultProblem: Problem = problemOf(StatusCodes.BadRequest, UpdatePurposeBadRequest(purposeId))
-    onComplete(result) {
-      handleApiError(defaultProblem) orElse handleOrganizationRoleError orElse {
-        case Success(purpose)                          =>
-          updatePurpose200(purpose)
-        case Failure(ex: RiskAnalysisValidationFailed) =>
-          logger.error(s"Error Updating Purpose $purposeId - Risk Analysis Validation failed", ex)
-          val problem = problemOf(StatusCodes.BadRequest, RiskAnalysisFormError(ex.getMessage))
-          createPurpose400(problem)
-        case Failure(ex)                               =>
-          logger.error(s"Error updating Purpose $purposeId", ex)
-          updatePurpose400(defaultProblem)
-      }
-    }
+    onComplete(result) { updatePurposeResponse[Purpose](operationLabel)(updatePurpose200) }
   }
 
   override def getPurpose(id: String)(implicit
     contexts: Seq[(String, String)],
     toEntityMarshallerPurpose: ToEntityMarshaller[Purpose],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = authorize(ADMIN_ROLE, M2M_ROLE) {
-    logger.info("Retrieving Purpose {}", id)
+    val operationLabel = s"Retrieving Purpose $id"
+    logger.info(operationLabel)
+
     val result: Future[Purpose] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       uuid           <- id.toFutureUUID
       purpose        <- purposeManagementService.getPurpose(uuid)
       eService       <- catalogManagementService.getEServiceById(purpose.eserviceId)
+      // TODO Purposes should be accessible to everyone? This would fail if org is not Consumer or Producer
       ownership      <- Ownership
         .getOrganizationRole(organizationId, eService.producerId, purpose.consumerId)
         .toFuture
```

Comment:
purpose retrieve should be allowed also to tenants that are not the consumer or the producer, but this expression returns an error if the requester is not one of the two.
This is used only for response decoration, that should be removed in the near future in favour of BFF.

The first workaround that came to mind is to default to PRODUCER, which in enhancement allows to return the most restricted dataset. But I don't like it too much because it hides the behaviour.
Or I could just to it and add a comment.

Opinions?

Reply:
I got the point and agree that this part will be removed with the implementation on the BFF side.
I don't have a better solution so a comment would be helpful together with the default value.